src/nlp/arrow_writer.py

```diff
@@ -100,10 +102,37 @@ def _write_array_on_file(self, pa_array):
         self._num_bytes += pa_array.nbytes
         self.pa_writer.write_batch(pa_batch)
 
+    def _sort_fields(self, datatype):
+        sorted_fields = sorted(datatype, key=lambda x: x.name)
+        sorted_names = [x.name for x in sorted_fields]
+        return sorted_names, sorted_fields
+
     def write_on_file(self):
         """ Write stored examples
         """
-        if self.current_rows:
+        ext_cols = set(map(lambda x: x.name, (filter(
+            lambda x: isinstance(x.type, pa.PyExtensionType),
+            self._type
+        ))))
+        # pyarrow.lib.ArrowNotImplementedError: Sequence converter
+        # for type extension<arrow.py_extension_type> not implemented
+        # so, we must sequence them ourselves
+        if ext_cols and self.current_rows:
+            entries = []
+            for row in self.current_rows:
+                row_list = list(
+                    map(lambda x: pa.array([row[x]], self._type[x].type)
+                        if x not in ext_cols else row[x], self._sorted_names))
```

Comment:
same here
```suggestion
                row_list = [pa.array([row[x]], self._type[x].type)
                        if x not in ext_cols else row[x] for x in self._sorted_names]
```

Reply:
also, why not have [row[x]] for extension types too ?

Reply:
ah, the only reason i don't encapsulate extension types within a list, is because I already do so in the encode_examples() method for Array2D. I would have included it here, but I needed to add an extra dimension to the the example data before it gets casted to an Extension data type.